release.py

```diff
@@ -0,0 +1,694 @@
+            continue
+
+        if not re.match(r'^\d+\.\d+\.\d+\.dev\d+$', suggested_version):
+            logger.error('Error: Version must be in format X.Y.Z.devN')
+            continue
+
+        confirm = (
+            input(
+                f"Confirm using scenario version '{suggested_version}' for post release? [y/N]: "
+            )
+            .strip()
+            .lower()
+        )
+
+        if confirm == 'y':
+            break
+
+        logger.info("Let's try again...")
+
+    return suggested_version
+
+
+def update_versions_for_post_release(repo: github.Repository.Repository, branch_name: str):
+    """Update version files to the post-release version with '.dev0' suffix."""
+    ops_version = get_new_version_post_release(repo, branch_name)
+    scenario_version = get_new_scenario_version_post_release()
+    update_ops_version(ops_version, scenario_version)
+    update_testing_version(ops_version, scenario_version)
+    update_tracing_version(ops_version)
+    update_uv_lock()
+
+
+def check_update_charm_pins_prs(repo: github.Repository.Repository):
+    """Check for open PRs that update charm pins."""
+    prs = repo.get_pulls(state='open')
+    open_prs = [pr for pr in prs if pr.title == 'chore: update charm pins']
+    if open_prs:
+        logger.info('Please merge "update charm pins" PRs first:')
+        pr = open_prs[0]
+        pr_url = f'#{pr.number} - {pr.html_url}'
+        logger.info(f'  {pr_url}')
+        if len(open_prs) > 1:
+            logger.info(
+                f'Note that there are {len(open_prs) - 1} more open update charm pins PRs.'
+            )
+
+
+def draft_release(owner: str, repo_name: str, base_branch: str):
+    """Create a draft release, update changelog, and create a PR for the release."""
+    org = gh_client.get_organization(owner)
+    repo = org.get_repo(repo_name)
+
+    check_update_charm_pins_prs(repo)
+
+    tag = get_new_tag_for_release(owner, repo, base_branch)
+    release = create_draft_release(repo, tag, base_branch)
+    if not release:
+        logger.error('Failed to create draft release.')
+        exit(1)
+    logger.info(f'Draft release created: {release.html_url}.')
+
+    categories, full_changelog = parse_release_notes(release.body)
+    notes = format_release_notes(categories, full_changelog)
+    print_release_notes(notes)
+
+    title, summary = input_title_and_summary(release)
+    if not title:
+        title = tag
+    notes = f'{summary}\n{notes}'
+
+    update_draft_release(release, title, notes)
+
+    changes = format_changes(categories, tag)
+    update_changes_file(changes, 'CHANGES.md')
+
+    update_versions_for_release(tag)
+
+    new_branch = f'release-prep-{tag}'
+    subprocess.run(['/usr/bin/git', 'checkout', '-b', new_branch], check=True)
+    changed_files = ['CHANGES.md', *[str(path) for path in VERSION_FILES.values()]]
+    for file in changed_files:
+        subprocess.run(['/usr/bin/git', 'add', file], check=True)
```

Comment:
Could do it in one operation instead, something like:
```suggestion
    subprocess.run(['/usr/bin/git', 'add', *changed_files], check=True)
```